pkg/controller/controller_instance_test.go

```diff
@@ -739,6 +797,37 @@ func TestReconcileInstanceDeleteDoesNotInvokeBroker(t *testing.T) {
 	assertNumEvents(t, events, 0)
 }
 
+func TestReconcileInstanceWithFailureCondition(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstanceWithFailedStatus()
+
+	if err := testController.reconcileInstance(instance); err != nil {
+		t.Fatalf("This should not fail : %v", err)
+	}
+
+	brokerActions := fakeBrokerClient.Actions()
+	assertNumberOfBrokerActions(t, brokerActions, 0)
+
+	if testController.pollingQueue.Len() != 0 {
+		t.Fatalf("Expected the polling queue to be empty")
+	}
+
+	actions := fakeCatalogClient.Actions()
+	assertNumberOfActions(t, actions, 0)
+
+	// verify no kube resources created.
+	// One single action comes from getting namespace uid
```

Comment:
Comment says one action followed by a check for zero actions. What does that mean?